independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java

```diff
@@ -112,23 +122,30 @@ public void execute() throws MojoExecutionException {
         }
 
         // extension.json
-        ObjectMapper mapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+        if (extensionFile == null) {
+            extensionFile = new File(outputDirectory,
+                    "META-INF" + File.separator + BootstrapConstants.QUARKUS_EXTENSION_FILE_NAME);
+        }
+
         ObjectNode extObject;
-        if (extensionJson == null) {
-            extensionJson = new File(outputDirectory,
-                    "META-INF" + File.separator + BootstrapConstants.EXTENSION_PROPS_JSON_FILE_NAME);
+        if (!extensionFile.exists()) {
+            // if does not exist look for fallback .json
+            extensionFile = new File(extensionFile.getParent(), "quarkus-extension.json");
```

Comment:
ultimately I would like to have the extensionFile be null and then look for yaml first, json second but didn't find api